typed/parser.go

```diff
@@ -68,34 +68,36 @@ func (p *Parser) TypeNames() (names []string) {
 
 // Type returns a helper which can produce objects of the given type. Any
 // errors are deferred until a further function is called.
-func (p *Parser) Type(name string) *ParseableType {
-	return &ParseableType{
+func (p *Parser) Type(name string) ParseableType {
+	return &parseableType{
 		parser:   p,
 		typename: name,
 	}
 }
 
 // ParseableType allows for easy production of typed objects.
-type ParseableType struct {
+type ParseableType interface {
+	IsValid() bool
+	FromYAML(YAMLObject) (TypedValue, error)
+	FromUnstructured(interface{}) (TypedValue, error)
+}
+
+type parseableType struct {
```

Comment:
Maybe this could be named, schemaParseableType, explicitParseableType or something else (neither of those sound too good to me), to indicate that this one comes from a schema?

Reply:
Yeah, it's private though so it's less critical?